OpenRA.Mods.Common/Traits/AffectsShroud.cs

```diff
@@ -64,12 +65,20 @@ public void Tick(Actor self)
 			var projectedPos = centerPosition - new WVec(0, centerPosition.Z, centerPosition.Z);
 			var projectedLocation = self.World.Map.CellContaining(projectedPos);
 			var disabled = IsDisabled(self);
+			var traitDisabled = IsTraitDisabled;
+
+			if (traitDisabled)
+			{
+				cachedTraitDisabled = traitDisabled;
+				return;
+			}
 
 			if (cachedLocation == projectedLocation && cachedDisabled == disabled)
```

Comment:
Uh, but if `traitDisabled` is `false` and `cachedTraitDisabled` is `true`, we do *not* want to return early, but rather update, right?